src/calls.rs

```diff
@@ -28,6 +33,41 @@ impl Calls {
     pub fn iter(&self) -> impl Iterator<Item = &Call> {
         self.calls.iter()
     }
+
+    // appends `other` into a new [`Calls`].
+    pub fn extend_from_other(&mut self, other: Self) {
+        self.calls.extend(other.calls.into_iter());
+    }
+
+    // Consumes self and returns a new [`Calls`] with offset calls filtered out.
+    pub fn filter_offset_calls(self) -> Self {
```

Comment:
`Vec::retain` maybe?

Reply:
I knew there was something that did that. Updated in b68e35c